libpromises/generic_agent.c

```diff
@@ -989,6 +990,31 @@ void CloseLog(void)
 {
     closelog();
 }
+bool SetGroupToSystemGroup()
+{
+    // If effective user is root then change effective group to CF_SYSTEM_GROUP
+    // otherwise leave group as-is.
+    uid_t p_euid = geteuid();
+    if (p_euid == 0)
```

Comment:
I think this could be shortened to `if (geteuid() == 0)`, but YMMV.

Reply:
ye, both are fine IMO.